icc-x-api/utils/efact-util.ts

```diff
@@ -268,8 +270,17 @@ function toInvoice(
   // FIXME : coder l'invoice ref
   invoice.invoiceRef = uuidBase36(invoiceDto.id!!)
   invoice.ioCode = insurance.code!!.substr(0, 3)
+
+  //  Here we set the healthcare provider identification (nihii) based in the invocing code responsible (the creator of the Invoice code)
+  //  Doing so the nihiiHealthcareProvider becomes useless as the healthcare provider identification is exclusively based on the invoicing code
+  //  Why not using the Invoice responsible in the call of toInvoice instead ?
   invoice.items = _.map(invoiceDto.invoicingCodes, (invoicingCodeDto: InvoicingCodeDto) => {
-    return toInvoiceItem(nihiiHealthcareProvider, patientDto, invoiceDto, invoicingCodeDto)
+    return toInvoiceItem(
+      (hcpNihiiByIds || {})[invoiceDto.responsible!!],
```

Comment:
This logic will not work for interns.
This will set the `doctorIdentificationNumber` to the `nihii` of the the intern while we need the `maitre de stage`

I think this will work:
```` typescript
return toInvoiceItem(
      invoiceDto.supervisorNihii,
      patientDto,
      invoiceDto,
      invoicingCodeDto
)
````

When the `intern` does not select a supervisor
 * `supervisorNihii` will be his/her `nihii`
 * `internNihii` will not be set

When the `intern` does select a supervisor:
 * `supervisorNihii` will be the `nihii` of the `maitre de stage`
 * `internNihii` will be set to his/her `nihii`

When no intern is involved:
 * `supervisorNihii` will the `nihii` of the hcp
 * `internNihii` will not be set

Reply:
In the component ``, the `onSave` function does:

````typescript
        let invoices = _invoices.map(_invoice => {
          const invoice = new Invoice(_invoice);
          invoice.internship = this.internship;
          invoice.supervisor = this.hcparty;
          invoice.gnotion = dmgManager;
          invoice.longDelayJustification = this.longDelayJustification;
          invoice.date = date;
          invoice.groupId = groupId;
          return invoice;
        });
````

`internship` and `supervisor` are setters, defined in the model:
````typescript
  set supervisor(hcp: HealthcarePartyDto) {
    this._invoice.supervisorNihii = _.get(hcp, 'nihii');
    this._invoice.supervisorSsin = _.get(hcp, 'ssin');
    this._invoice.supervisorLastName = _.get(hcp, 'lastName');
    this._invoice.supervisorFirstName = _.get(hcp, 'firstName');
    this._invoice.supervisorCbe = _.get(hcp, 'cbe');
    // TODO correct mapping of professions
    this._invoice.supervisorCdHcParty = hcp ? 'persphysician' : undefined;
  }

  set internship(hcp: HealthcarePartyDto) {
    this._invoice.internshipNihii = _.get(hcp, 'nihii');
    this._invoice.internshipSsin = _.get(hcp, 'ssin');
    this._invoice.internshipLastName = _.get(hcp, 'lastName');
    this._invoice.internshipFirstName = _.get(hcp, 'firstName');
    this._invoice.internshipCbe = _.get(hcp, 'cbe');
    // TODO correct mapping of professions
    this._invoice.internshipCdHcParty = hcp ? 'persphysician' : undefined;
  }
````

Reply:
do not pass `nihiiHealthcareProvider: string` in `toInvoiceItem` as it is not used anymore